src/main/java/io/jenkins/plugins/services/impl/HttpClientGithubReleases.java

```diff
@@ -0,0 +1,68 @@
+package io.jenkins.plugins.services.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import io.jenkins.plugins.endpoints.PluginEndpoint;
+import io.jenkins.plugins.models.Plugin;
+import io.jenkins.plugins.models.PluginRelease;
+import io.jenkins.plugins.models.PluginReleases;
+import io.jenkins.plugins.services.ConfigurationService;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+public class HttpClientGithubReleases extends HttpClient {
+  private final ConfigurationService configurationService;
+  private Logger logger = LoggerFactory.getLogger(PluginEndpoint.class);
+
+  public HttpClientGithubReleases(ConfigurationService configurationService) {
+    super();
+    this.configurationService = configurationService;
+  }
+
+
+  public PluginReleases getReleases(Plugin plugin) throws IOException {
+    String clientId = this.configurationService.getGithubClientId();
+    String clientSecret = this.configurationService.getGithubClientSecret();
+
+    if (clientId == null) {
+      logger.warn("No GitHub Client ID specified, using anonymous credentials to access github api");
+    }
+
+    final String[] scmUrl = StringUtils.removeEnd(
+      plugin.getScm().getLink().replaceAll("https://github.com/", "").replaceAll( "http://github.com/", ""),
+      "/"
+    ).split("/");
+
+    if (scmUrl.length != 2) {
+      logger.warn("Not sure what to do with SCM URL of " + plugin.getScm().getLink() + " == " + Arrays.toString(scmUrl));
+      return new PluginReleases(new ArrayList<>());
+    }
+
+    final String URL = String.format(
```

Comment:
Is it worth adding this to the GitHub-api library instead? If it’s not already there?

Reply:
The GitHub API isn't included in the he pom.

I tried installing the API. It needs a newer version of Jackson. Newer version of Jackson breaks elastic search. i gave up then.